### ISSUE TITLE
Validation & Early Stopping in  Scikit-Learn API

### ISSUE THREAD

**Issue by jskDr** [fchollet/keras#3400]
How can I add early stopping step when I use **wrappers for the Scikit-Learn API**? I found a tutorial related to the Scikit-Learn API but there is no description of it. If there is no way to do early stopping, it is a good but not useful for many practical cases. Especially when the number of weights are more than the number of data such as not large data cases, over-fitting diminish the performance of learning. I know there is other techniques to prohibit over-fitting such as L1/L2 regularization, dropout. However, no supporting validation in a fitting stage prevents monitoring a practical testing performance but not a ideal learning performance. 
http://machinelearningmastery.com/regression-tutorial-keras-deep-learning-library-python/#comment-358900
